src/zap/zapimport.cpp

```diff
@@ -1773,6 +1775,40 @@ void ZapImportSectionSignatures::PlaceDynamicHelperCell(ZapImport * pImport)
     m_pImage->GetImportTable()->PlaceImportBlob(pCell);
 }
 
+ZapImport * ZapImportTable::GetDictionaryLookupCell(CORCOMPILE_FIXUP_BLOB_KIND kind, CORINFO_RESOLVED_TOKEN * pResolvedToken, CORINFO_LOOKUP_KIND * pLookup)
+{
+    _ASSERTE(pLookup->needsRuntimeLookup);
+
+    SigBuilder sigBuilder;
+
+    sigBuilder.AppendData(kind & ~CORINFO_HELP_READYTORUN_ATYPICAL_CALLSITE);
+
+    switch (pLookup->runtimeLookupFlags)
+    {
+    case READYTORUN_FIXUP_TypeHandle:
+        {
+            if (pResolvedToken->pTypeSpec == NULL)
+            {
+                _ASSERTE(!"Invalid IL that directly references __Canon");
+                ThrowHR(E_NOTIMPL);
+            }
+
+            sigBuilder.AppendData(ENCODE_TYPE_HANDLE);
+            sigBuilder.AppendData(pResolvedToken->cbTypeSpec);
```

Comment:
This should be encoded in the exact same way as the non-generic `ENCODE_TYPE_HANDLE`. There is no number of bytes in the non-generic `ENCODE_TYPE_HANDLE` - it is immediately followed by the typespec.